### PR TITLE
Merge -amo config files into bare config files

### LINKED ISSUE #9832: Merge -amo config files into bare config files
This supports https://github.com/mozilla/addons-frontend/pull/9830

Once we eliminate the `NODE_APP_INSTANCE` env var, we will no longer be loading two config files for addons-frontend. Currently, because of https://github.com/lorenwest/node-config/wiki/Multiple-Node-Instances, we load two config files. For example, for development we load both `development.js` and `development-amo.js`. It looks like the current behaviour is for the values in the `-amo` file to overwrite any values from the bare file, so we should keep that in mind when merging the two.

This change should land before the changes for #9830

### PR DESCRIPTION
**Pull request by bobsilverberg**
Fixes #9832

### CODE REVIEW

**Comment by bobsilverberg** [comment]
@willdurand This passes locally, but fails on CI, because of the CDN URLs, I believe. We have `addonsServerProdCDN`, which is `https://addons.cdn.mozilla.net`, and we also have `addonsFrontendCDN`, which is `https://addons-amo.cdn.mozilla.net`. When I merged the files, some things that used to point to `addonsServerProdCDN` now point to `addonsFrontendCDN`. These include `staticHost` as well as a bunch of CSP directives. Does this have to do with environment variables, and do those need to be changed? Also, do we still need both `https://addons.cdn.mozilla.net` and `https://addons-amo.cdn.mozilla.net`?

File: config/default.js [hunk 1]
```diff
@@ -157,24 +158,31 @@ module.exports = {
   // from ./lib/shared.js
   CSP: {
     directives: {
-      defaultSrc: ["'none'"],
+      defaultSrc: [addonsFrontendCDN],
```
Comment (willdurand) [inline config/default.js:161]: this one needs to be above `prefetchSrc` because the comment applies to it

**Comment by willdurand** [comment]
> @willdurand This passes locally, but fails on CI, because of the CDN URLs, I believe.

The problem with the current test suite is that it does not really care about the `-amo` config. It uses the `test.js` config in addition to `default.js`. Those test errors are correct and we need to fix the test cases now..

> We have `addonsServerProdCDN`, which is `https://addons.cdn.mozilla.net`, and we also have `addonsFrontendCDN`, which is `https://addons-amo.cdn.mozilla.net`.

Yes, we have a CDN for the addons-frontend compiled files (deployments) and the CDN for the AMO platform itself (i.e. addons-server).

> When I merged the files, some things that used to point to `addonsServerProdCDN` now point to `addonsFrontendCDN`. These include `staticHost` as well as a bunch of CSP directives.

Yeah, and that looks good to me.

> Does this have to do with environment variables, and do those need to be changed?

I don't think we need to change anything here.

> Also, do we still need both `https://addons.cdn.mozilla.net` and `https://addons-amo.cdn.mozilla.net`?

Yes

**Comment by bobsilverberg** [comment]
@willdurand I managed to fix most of the errors (which, contrary to what I said earlier, were reproducing locally), but there are 3 left that I cannot seem to figure out:

`tests/unit/core/server/test_base.js › app › returns the status code of NestedStatus`
- I can't figure out why this is returning a 301 rather than a 404
- also, this test actually shows up as failing twice, once as part of `tests/unit/core/server/test_base.js` and once as part of `tests/unit/core/middleware/test_datadogTiming.js`. I don't really understand that either.

`tests/unit/core/components/TestErrorPage.js › catches and reports errors in child components`
- I cannot reproduce this locally, even if I get rid of the `test.js` config file, and I'm not sure what's causing the failure.

If you have some time and can take a look at these I'd appreciate it.

**Comment by willdurand** [comment]
> `tests/unit/core/server/test_base.js › app › returns the status code of NestedStatus`

This one is because `enableTrailingSlashesMiddleware` now defaults to `true`. Adding a slash like this should fix the problem:

```diff
diff --git a/tests/unit/core/server/test_base.js b/tests/unit/core/server/test_base.js
index bd8ad5355..38a96dfed 100644
--- a/tests/unit/core/server/test_base.js
+++ b/tests/unit/core/server/test_base.js
@@ -169,7 +169,7 @@ describe(__filename, () => {
       );
 
       const response = await testClient({ App: NotFoundApp })
-        .get('/en-US/firefox/simulation-of-a-non-existent-page')
+        .get('/en-US/firefox/simulation-of-a-non-existent-page/')
         .end();
 
       expect(response.statusCode).toEqual(404);
```

**Comment by willdurand** [comment]
> also, this test actually shows up as failing twice, once as part of `tests/unit/core/server/test_base.js` and once as part of `tests/unit/core/middleware/test_datadogTiming.js`. I don't really understand that either.

Likely caused by the test case in `test_base.js`. Fixing the former should fix both issues.

> tests/unit/core/components/TestErrorPage.js › catches and reports errors in child components

I cannot reproduce either. Maybe also caused by the first error?

**Comment by bobsilverberg** [comment]
Thanks @willdurand, good call on the trailing slash thing! I've fixed those two failures, but the one for `TestErrorPage.js` still exists. I don't see anything in the test that would suggest it has anything to do with a slash. I notice the problem is that it expects our user-defined error to be loaded, but instead an error for `Cannot read property 'location' of null` gets loaded, so I guess that is somehow getting triggered unexpectedly. It's quite strange that it doesn't reproduce locally though.

I'll do some more digging, but it's pretty tough to debug when I cannot recreate the failure. Let me know if you have any other ideas.

**Comment by willdurand** [comment]
> Thanks @willdurand, good call on the trailing slash thing! I've fixed those two failures, but the one for `TestErrorPage.js` still exists. I don't see anything in the test that would suggest it has anything to do with a slash. I notice the problem is that it expects our user-defined error to be loaded, but instead an error for `Cannot read property 'location' of null` gets loaded, so I guess that is somehow getting triggered unexpectedly. It's quite strange that it doesn't reproduce locally though.
> 
> I'll do some more digging, but it's pretty tough to debug when I cannot recreate the failure. Let me know if you have any other ideas.

@bobsilverberg right, I didn't pay too much attention to the remaining one but I think it's because Sentry is enabled. This is because `publicSentryDsn` isn't `null` anymore so I suggest this fix:

```diff
commit d749bc6583ddf60d8ab8534f14c2cbb0e995a6ad
Author: William Durand <will+git@drnd.me>
Date:   Thu Nov 5 15:39:33 2020 +0100

    disable sentry

diff --git a/config/test.js b/config/test.js
index e0c6323fc..3342e46eb 100644
--- a/config/test.js
+++ b/config/test.js
@@ -14,6 +14,9 @@ module.exports = {
   // memory we have (and more...).
   enableRequestID: false,
 
+  // Force-disable Sentry
+  publicSentryDsn: null,
+
   // We use a fake/incomplete file for the test suite.
   loadableStatsFile: path.join(fixturesPath, 'loadable-stats.json'),

```

**Comment by bobsilverberg** [comment]
Everything is passing now! Thanks for your help, @willdurand. :-)

**Review by willdurand — APPROVED** [review]

**Comment by bobsilverberg** [comment]
Hmm, something went wrong during a rebase, and the final patch didn't remove dev-amo or development-amo. I'll file a follow-up.

**Pull request merged**